src/gt4py/next/program_processors/runners/dace/gtir_dataflow.py

```diff
@@ -835,12 +837,19 @@ def write_output_of_nested_sdfg_to_temporary(inner_value: ValueExpr) -> ValueExp
 
         outputs = {outval.dc_node.data for outval in gtx_utils.flatten_nested_tuple((result,))}
 
+        if nsdfg_symbols_mapping is None:
+            # `None` means that all free symbols are mapped to the symbols available in parent SDFG
```

Comment:
```suggestion
            # `None` means that all free symbols are mapped to the symbols available in parent SDFG by the `add_nested_sdfg()` function.
```
This case also means that we never need to do a remapping where the names inside and outside are different.

I would actually kill this `if` because as far as I can see `nsdfg_symbol_mapping` is either `None` or `{"__cond": ...}`.

Reply:
Bump

Reply:
Makes sense, I will simplify this code. However, I have to deal with the mapping of the domain symbols:
`ValueError: Missing symbols on nested SDFG "if_stmt_0": ['i_KDim_gtx_vertical', 'i_IDim_gtx_horizontal']`